CONTRIBUTING.md

```diff
@@ -2,11 +2,75 @@
 
 The github actions in this repo are used in several projects.
 
-The projects mentioned in [README.md](README.md) are considered critical users of this repo. When introducing a change in this repo, 
-care should be taken to check that those will not break the CI process of the critical users.
-
-The verification process is as follow:
-- Create a branch with the changes in the GitHub actions. Push this branch to the GitHub repo.
-- Create branches out of current master in the projects mentioned in [README.md](README.md). In those branches, point the GitHub actions to your new branch.
-- Create pull requests, marked as `do-not-merge` in the projects so that CI is triggered with your new changes.
-- Once working, submit a PR to the GitHub actions repo.
+The projects mentioned in [README.md](README.md) are considered critical users of this repo.
+When introducing a change in this repo, care should be taken to check that those changes will not break the CI process of the critical users.
+
+To submit a change please follow the guidelines described below.
+
+## Basics
+### Pointing to a specific action
+The syntax for using a public action in a subdirectory is as follows ([reference](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-public-action-in-a-subdirectory)):
+```
+uses: {owner}/{repo}/{path}@{ref}
+```
+
+For example:
+```
+uses: mfem/github-actions/build-hypre@v2.4
+```
+`{ref}` can be *any* of the following:
+- branch
+- tag
```

Comment:
I would add a note to them indicating how we prefer to use them, just as a small outlook. Something like "branch - (recommended for development)", "tag - (recommended for final version)".